### PR TITLE
PIPE-1072-2: fix explicitly setting job default queue causing controller to miss the job

### PR DESCRIPTION
**Pull request by zhming0**
When bk yaml looks like: 

```
agent: 
  queue: default # -> happen to be the default queue
```

And k8s controller starts without a queue tag, the controller will miss the job because controller wants a job that has `queue=""` instead. 

----

This PR fixed it on transport layer so when we poll jobs, its queue tag will always match our expectation.

### CODE REVIEW

File: internal/controller/agenttags/tags.go [hunk 1]
```diff
@@ -42,6 +42,16 @@ func SetTag(tags []string, key, val string) []string {
...
 }
 
+func RemoveTag(tags []string, key string) []string {
+	i := slices.IndexFunc(tags, func(tag string) bool {
+		return strings.HasPrefix(tag, key+"=")
+	})
+	if i < 0 {
+		return tags
+	}
+	return slices.Delete(tags, i, i+1)
```
Comment (DrJosh9000) [inline internal/controller/agenttags/tags.go]: Avoids dealing with an index `i`:

```suggestion
	return slices.DeleteFunc(tags, func(tag string) bool {
		return strings.HasPrefix(tag, key+"=")
	})
```

File: internal/integration/integration_test.go [hunk 2]
```diff
@@ -71,6 +71,37 @@ func TestDefaultQueue(t *testing.T) {
 	tc.AssertLogsContain(build, "Hi there")
 }
 
+// This is test case for an interesting edge case!
+// Read more from: https://buildkite-corp.slack.com/archives/C08N9Q51CTC/p1749704326621419?thread_ts=1749704031.661719&cid=C08N9Q51CTC
```
Comment (DrJosh9000) [inline internal/integration/integration_test.go]: Rather than a link to Slack, a couple of words explaining it here would be better both for others (that don't have access to our Slack instance) and us (so we don't have to read through the thread, some of which isn't relevant to  this bug).

Or maybe the comment in `GetScheduledJobs` is enough?

**Review by DrJosh9000 — APPROVED** [review]

File: internal/integration/integration_test.go (hunk 2 above)
Comment (zhming0) [reply internal/integration/integration_test.go]: ah yeah this is a public repo! 🤦🏿 

I removed it. I agree git blame and other comments should offer enough breadcrumbs.
